### ISSUE TITLE
Unexpected behavior with header styling

### ISSUE THREAD

**Issue by jakeleventhal** [mbrn/material-table#328]
**Describe the bug**
https://mbrn.github.io/material-table/#/docz-examples-10-example-component-overriding
When trying to apply this to the table header, the cell columns get unaligned

**Screenshots**
Here is my code with a screenshot:
```
components={{
                    Header: props => (
                      <div style={{ background: palette.primary.main }}>
                        <MTableHeader {...props} />
                      </div>
                    )
                  }}
```
<img width="1055" alt="Screen Shot 2019-03-15 at 11 54 48 AM" src="https://user-images.githubusercontent.com/15133828/54444497-37f3f480-4719-11e9-9bda-23068ee8828a.png">

**Comment by mbrn** [comment]
Hi @jakeleventhal ,

Can you try this instead of it:

```javascript
<MaterialTable
  options={{
    headerStyle: { background: 'red' }
  }}
  columns={[
    { title: 'Adı', field: 'name' },
    { title: 'Soyadı', field: 'surname' },
    { title: 'Doğum Yılı', field: 'birthYear', type: 'numeric' },
    {
      title: 'Doğum Yeri',
      field: 'birthCity',
      lookup: { 34: 'İstanbul', 63: 'Şanlıurfa' },
    },
  ]}
  data={[
    { name: 'Mehmet', surname: 'Baran', birthYear: 1987, birthCity: 63 },
  ]}
  title="Demo Title"
/>
```

**Comment by jakeleventhal** [comment]
That works, the docs made it a little hard to infer that this was the correct solution though.

Thanks!

**Comment by jakeleventhal** [comment]
This change breaks when upgrading from 1.24->1.25+

**Comment by mbrn** [comment]
Hi @jakeleventhal ,

It is fixed with 1.28.1.
